File: Hotelaria/src/main/java/com/dolts/hotelaria/dto/RequestDTO.java
package com.dolts.hotelaria.dto;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.util.List;

@Getter
@Setter
public class RequestDTO implements Serializable {
    private List<SearchRequestDTO> searchRequestDTOS;
    private GlobalOperator globalOperator;
    private PageRequestDTO pageDTO;

    public enum GlobalOperator {
        AND, OR;
    }
}

File: hotelaria-frontend/src/app/shared/dto/requestDTO.ts
import {PageDTO} from "./pageDTO";
import {SearchRequestDTO} from "./searchRequestDTO";

export class RequestDTO {
    searchRequestDTOS: SearchRequestDTO[] = [];
    globalOperator: GlobalOperator;
    pageDTO: PageDTO;
}

export enum GlobalOperator {
    AND = 'AND', OR = 'OR'
}
